multiqc/modules/atropos/atropos.py

```diff
@@ -0,0 +1,1356 @@
+    
+    def __repr__(self):
+        return self.name
+
+FAIL = Status('fail', 0, '#d9534f')
+WARN = Status('warn', 1, '#f0ad4e')
+PASS = Status('pass', 2, '#5cb85c')
+
+DEFAULT_COLOR = '#999'
+
+ADD_LISTENERS = '<script type="text/javascript">$(function () {{ add_atropos_listeners("{phase}", atropos_passfails_{phase})  }});</script>'
+PASSFAILS = '<script type="text/javascript">atropos_passfails_{phase} = {statuses};</script>'
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+## Module
+
+class MultiqcModule(BaseMultiqcModule):
+    """Base Atropos module class. Loads JSON ouptut. Submodule classes are
+    responsible for extracting data and generating summary table and plots.
+    
+    Args:
+        name: Module name.
+        anchor: HTML anchor name.
+        from_config: Whether to load data from configured locations. This should
+            only be set to False in tests.
+    """
+    def __init__(self, name='', anchor='', info='', from_config=True):
+        # Create the submodule based on the phase
+        mod_cust_config = getattr(self, 'mod_cust_config', {})
+        phase = mod_cust_config.get('phase', 'trim')
+        submodule_class = SUBMODULES[phase]
+        self.submodule = submodule_class()
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name=name or self.submodule.name,
+            anchor=anchor or self.submodule.anchor,
+            href=ATROPOS_GITHUB_URL,
+            info=info or self.submodule.info + (
+                "is a general-purpose NGS pre-processing tool that "
+                 "specializes in adatper- and quality-trimming, written "
+                 "by John Didion at NHGRI. It is a fork of Cutadapt, "
+                 "written by Marcel Martin at Stockholm University."))
+
+        self.css = {}
+        self.js = {}
+
+        # Load data from logs and generate the report
+        if from_config:
+            self.init_from_config()
+            self.submodule.atropos_report(self)
+
+    def __getattr__(self, name):
+        """Fetch all missing attributes from the submodule.
+        """
+        return getattr(self.submodule, name)
+
+    def init_from_config(self):
+        """Initialize module from JSON files discovered via MultiQC
+        configuration.
+        """
+        log_files = self.find_log_files('atropos', filehandles=True)
+        if not log_files:
+            # Atropos summary files are JSON files (with .json extension) and
+            # always have '"program": "Atropos"' as the first key-value pair.
+            # Note that this way is deprecated; currently it produces a warning
+            # but it may produce an error in the future.
+            patterns = dict(fn='*.json', contents='"program": "Atropos"')
+            log_files = self.find_log_files(patterns, filehandles=True)
+
+        for file_dict in log_files:
+            fileobj = file_dict['f']
+            data = json.load(fileobj)
```

Comment:
I recommend putting this in a `try`/ `except` block - you'd be amazed how many people have truncated and corrupted files which can break json parsing...

Reply:
done